pkg/core/secrets/manager/validator.go

```diff
@@ -0,0 +1,72 @@
+package manager
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/pkg/errors"
+
+	mesh_proto "github.com/Kong/kuma/api/mesh/v1alpha1"
+	"github.com/Kong/kuma/pkg/core/ca"
+	mesh_core "github.com/Kong/kuma/pkg/core/resources/apis/mesh"
+	"github.com/Kong/kuma/pkg/core/resources/model"
+	core_store "github.com/Kong/kuma/pkg/core/resources/store"
+	"github.com/Kong/kuma/pkg/core/validators"
+)
+
+// MeshAccessor is used instead of MeshManager, otherwise we would have cyclic dependency.
+// MeshManager depends on SecretManager which depends on SecretValidator which would depend on ResourceManager
+type MeshAccessor = func(context.Context, model.Resource, ...core_store.GetOptionsFunc) error
+
+type SecretValidator interface {
+	ValidateDelete(ctx context.Context, secretName string, secretMesh string) error
+}
+
+func NewSecretValidator(caManagers ca.Managers, meshAccessor MeshAccessor) SecretValidator {
+	return &secretValidator{
+		caManagers:   caManagers,
+		meshAccessor: meshAccessor,
+	}
+}
+
+type secretValidator struct {
+	caManagers   ca.Managers
+	meshAccessor MeshAccessor
+}
+
+func (s *secretValidator) ValidateDelete(ctx context.Context, secretName string, secretMesh string) error {
```

Comment:
Isn't it obvious that this is always `name` and `mesh` of the secret since we are in `secretValidator`? So parameters could be renamed to just `name` and `mesh`?

Reply:
I think I started with this, but then had a clash between `mesh` (a resource) and mesh (a name) so I changed mesh (name) to `meshName` and name to `secretName` for consistency, but I can go also with yours + `meshRes`